app/models/user.rb

```diff
@@ -69,6 +69,36 @@ def membership_current?
   end
 
 
+  # TODO this should not be the responsibility of the User class.
+  def membership_current_as_of?(this_date)
+    return false if this_date.nil?
+
+    membership_payment_expire_date = membership_expire_date
+    !membership_payment_expire_date.nil? && (membership_payment_expire_date > this_date)
+  end
+
+
+  # User has an approved membership application and
+  # is up to date (current) on membership payments
+  def membership_app_and_payments_current?
+    has_approved_shf_application? && membership_current?
+  end
+
+
+  # User has an approved membership application and
+  # is up to date (current) on membership payments
+  def membership_app_and_payments_current_as_of?(this_date)
+    has_approved_shf_application? && membership_current_as_of?(this_date)
+  end
```

Comment:
Hmmm ... I'd be a bit leery of testing date-dependent stuff and assuming that I don't have to freeze time simply because this particular method negates the need for that - for this particular aspect of date calculations.  And, event if the test works, it is subject to breakage if some other data calculation changes or is introduced elsewhere.

Reply:
Aside from my qualms about using this in testing, I can see where this might be useful in the future, in other contexts, so I come down on the side of "keep it".

Reply:
:-)  It's gone now.  But we can always revive it later.